### PR TITLE
Fix and improve array compare operations

### PR DESCRIPTION
**Pull request by qingshi163**

### CODE REVIEW

**Comment by coolreader18** [comment]
Weird, looks like there's a deadlock in an importlib test_deadlock function only on macos... I'll try rerunning the test and see if it was a fluke

**Comment by qingshi163** [comment]
Is it all right?

File: vm/src/stdlib/array.rs [hunk 1]
```diff
@@ -337,6 +339,18 @@ macro_rules! def_array_enum {
                     ret
                 })
             }
+
+            fn cmp(&self, other: &ArrayContentType) -> Result<Option<Ordering>, ()> {
+                match self {
+                    $(ArrayContentType::$n(v) => {
+                        if let ArrayContentType::$n(other) = other {
```
Comment (coolreader18) [inline vm/src/stdlib/array.rs:346]: You could also do something like

```
match (self, other) {
     (ArrayContentType::$n(v), ArrayContentType::$n(other)) => cmp(),
     _ => Err(())
}
```

**Review by coolreader18 — APPROVED** [review]
This looks great! I have one nit, otherwise this is good to merge!

**Pull request merged**